`ERR_PNPM_NO_MATCHING_VERSION` not showing error source info

I'm seeing this error printed:

```
packages/public/framework/ava            |   +3 +
No matching version found for babel-loader@>=8.0.0 <9.0.0
 ERROR  No matching version found for babel-loader@>=8.0.0 <9.0.0
```

In `pnpm-debug.log`:

```
  2119 error pnpm: 
    message: "No matching version found for babel-loader@>=8.0.0 <9.0.0"
    code: "ERR_PNPM_NO_MATCHING_VERSION"
    packageMeta: 
      versions: 
        4.0.0: 
          name: "babel-loader"
          version: "4.0.0"
          dependencies: 
			...
```

It doesn't provide the package (its a multi repo project) where the error was caused by. And I can't seem to find what is causing it.

## Solution

Add source of error information.

--- a/.changeset/shiny-years-brush.md
+++ b/.changeset/shiny-years-brush.md
@@ -1,0 +1,5 @@
+---
+"@pnpm/npm-resolver": patch
+---
+
+When no matching version is found, report the actually specified version spec in the error message (not the normalized one).

--- a/packages/npm-resolver/src/index.ts
+++ b/packages/npm-resolver/src/index.ts
@@ -23,7 +23,6 @@ import pickPackage, {
   PackageMetaCache,
   PickPackageOptions,
 } from './pickPackage'
-import toRaw from './toRaw'
 import path = require('path')
 import LRU = require('lru-cache')
 import normalize = require('normalize-path')
@@ -33,8 +32,11 @@ import ssri = require('ssri')
 
 class NoMatchingVersionError extends PnpmError {
   public readonly packageMeta: PackageMeta
-  constructor (opts: { spec: RegistryPackageSpec, packageMeta: PackageMeta}) {
-    super('NO_MATCHING_VERSION', `No matching version found for ${toRaw(opts.spec)}`)
+  constructor (opts: { wantedDependency: WantedDependency, packageMeta: PackageMeta}) {
+    const dep = opts.wantedDependency.alias
+      ? `${opts.wantedDependency.alias}@${opts.wantedDependency.pref ?? ''}`
+      : opts.wantedDependency.pref!
+    super('NO_MATCHING_VERSION', `No matching version found for ${dep}`)
     this.packageMeta = opts.packageMeta
   }
 }
@@ -149,7 +151,7 @@ async function resolveNpm (
       const resolvedFromLocal = tryResolveFromWorkspacePackages(workspacePackages, spec, opts.projectDir)
       if (resolvedFromLocal) return resolvedFromLocal
     }
-    throw new NoMatchingVersionError({ spec, packageMeta: meta })
+    throw new NoMatchingVersionError({ wantedDependency, packageMeta: meta })
   }
 
   if (workspacePackages?.[pickedPackage.name] && opts.projectDir) {

--- a/packages/npm-resolver/test/index.ts
+++ b/packages/npm-resolver/test/index.ts
@@ -792,7 +792,7 @@ test('error is thrown when there is no package found for the requested range', a
     await resolveFromNpm({ alias: 'is-positive', pref: '^1000.0.0' }, { registry })
     t.fail('installation should have failed')
   } catch (err) {
-    t.ok(err.message.startsWith('No matching version found for is-positive@>=1000.0.0 <1001.0.0'), 'failed with correct error message')
+    t.ok(err.message.startsWith('No matching version found for is-positive@^1000.0.0'), 'failed with correct error message')
     t.end()
   }
 })

